org.lflang/src/org/lflang/generator/ts/TSReactorGenerator.kt

```diff
@@ -0,0 +1,715 @@
+package org.lflang.generator.ts
+
+import org.lflang.*
+import org.lflang.ASTUtils.isInitialized
+import org.lflang.ASTUtils.toText
+import org.lflang.generator.FederateInstance
+import org.lflang.generator.PrependOperator
+import org.lflang.lf.*
+import org.lflang.lf.Timer
+import java.lang.StringBuilder
+import java.util.*
+import kotlin.collections.HashSet
+import kotlin.collections.LinkedHashMap
+
+/**
+ * Reactor generator for TypeScript target.
+ *
+ *  @author{Matt Weber <matt.weber@berkeley.edu>}
+ *  @author{Edward A. Lee <eal@berkeley.edu>}
+ *  @author{Marten Lohstroh <marten@berkeley.edu>}
+ *  @author {Christian Menard <christian.menard@tu-dresden.de>}
+ *  @author {Hokeun Kim <hokeunkim@berkeley.edu>}
+ */
+class TSReactorGenerator(
+    private val tsGenerator: TSGenerator,
+    private val errorReporter: ErrorReporter
+) {
+    private var code = StringBuilder()
+
+    /**
+     * Map from builder to its current indentation.
+     */
+    var indentation = LinkedHashMap<StringBuilder, String>()
+
+    // Wrapper functions and their helpers
+    private fun indent(builder: StringBuilder) = tsGenerator.indentw(builder)
+    private fun unindent(builder: StringBuilder) = tsGenerator.unindentw(builder)
+    private fun indent() = indent(code)
+    private fun unindent() = unindent(code)
+
+    private fun pr(builder: StringBuilder, text: Any) = tsGenerator.prw(builder, text)
+    private fun pr(text: Any) = tsGenerator.prw(code, text)
+
+    private fun getTargetValue(v: Value): String = tsGenerator.getTargetValueW(v)
```

Comment:
It would be more idiomatic to use extensions for those methods. You can write
```kotlin
 private fun Value.getTargetValue(): String = tsGenerator.getTargetValueW(v)
```

With this style, in my above comment,
 
```kotlin
                val timerPeriod = timer.period?.let { getTargetValue(it) } ?: "0"
```
can be rewritten to 

```kotlin
                val timerPeriod = timer.period?.getTargetValue() ?: "0"
```

Reply:
I tried as you commented, but it doesn't compile. Maybe I'm missing something?

Reply:
Look at the edit to my comment. If that doesn't work please post the compiler error

Reply:
Thanks, done in https://github.com/icyphy/lingua-franca/pull/444